### PR TITLE
verify that binaries have bitcode if necessary

### PR DESCRIPTION
**Pull request by jpsim**
/cc @tgoyne

### CODE REVIEW

**Comment by tgoyne** [comment]
:+1:

File: build.sh [hunk 1]
```diff
@@ -813,6 +817,36 @@ case "$COMMAND" in
...
+        if [[ ${#archs_array[@]} < 2 ]]; then
+            exit 1 # Early exit if not a fat binary
+        fi
+
+        TEMPDIR=$(mktemp -d $TMPDIR/realm-bitcode-check.XXXX)
+
+        for arch in $archs; do
+            lipo -thin "$arch" "$BINARY" -output "$TEMPDIR/$arch"
+            if otool -l "$TEMPDIR/$arch" | grep -q "segname __LLVM"; then
+                exit 0
```
Comment (bdash) [inline build.sh]: If I'm reading this right, we're only checking a single slice. Are we assuming that if one slice has bitcode that they all do? If so, why not check only the first architecture and return a result based on that?

Comment (jpsim) [inline build.sh]: simulator slices don't have bitcode. We could hardcode a list of architectures where we'd expect bitcode if you like.

Comment (bdash) [inline build.sh]: Doesn't seem worth it. A comment could help clarify the current behavior and reason for it.

**Comment by jakajancar** [comment]
@jpsim @bdash I believe this will detect `-fembed-bitcode-marker`, even if there is not bitcode (`-fembed-bitcode`).

**Comment by jpsim** [comment]
@jakajancar can you recommend an improvement?

**Pull request merged**